hudi-client/hudi-client-common/src/main/java/org/apache/hudi/table/HoodieTimelineArchiveLog.java

```diff
@@ -158,8 +158,8 @@ public boolean archiveIfRequired(HoodieEngineContext context) throws IOException
     // TODO (na) : Add a way to return actions associated with a timeline and then merge/unify
     // with logic above to avoid Stream.concats
     HoodieTimeline commitTimeline = table.getCompletedCommitsTimeline();
-    Option<HoodieInstant> oldestPendingCompactionInstant =
-        table.getActiveTimeline().filterPendingCompactionTimeline().firstInstant();
+    Option<HoodieInstant> oldestPendingCompactionAndReplaceInstant =
+        table.getActiveTimeline().filterPendingCompactionAndReplaceTimeline().firstInstant();
```

Comment:
possible to use `filterInflightsAndRequested()` instead of creating a new API `filterPendingCompactionAndReplaceTimeline()`?

Reply:
Sure thing, changed. Maybe use `!s.isComplete` is more accurate than `filterInflightsAndRequested()`.